src/core/connection.h

```diff
@@ -147,6 +147,12 @@ typedef union QUIC_CONNECTION_STATE {
         //
         BOOLEAN ResumptionEnabled : 1;
 
+        //
+        // Indicates that an app shutdown from a non worker thread is in progress.
+        // Received by the QUIC_CONNECTION_EVENT_SHUTDOWN_COMPLETE event.
+        //
+        BOOLEAN AppShutdownInProgress: 1;
```

Comment:
"Shutdown" is a different concept. It's in Close right now. Call it `AppCloseInProgress` instead.